In [9]:
from sklearn import datasets
from sklearn.model_selection import cross_val_score
from sklearn.naive_bayes import BernoulliNB, MultinomialNB, GaussianNB

In [5]:
digits_data = datasets.load_digits()
print(digits_data.data.shape)
print(digits_data.target.shape)
print(digits_data.target_names)
print(digits_data.data[:1])

(1797, 64)
(1797,)
[0 1 2 3 4 5 6 7 8 9]
[[ 0.  0.  5. 13.  9.  1.  0.  0.  0.  0. 13. 15. 10. 15.  5.  0.  0.  3.
  15.  2.  0. 11.  8.  0.  0.  4. 12.  0.  0.  8.  8.  0.  0.  5.  8.  0.
   0.  9.  8.  0.  0.  4. 11.  0.  1. 12.  7.  0.  0.  2. 14.  5. 10. 12.
   0.  0.  0.  0.  6. 13. 10.  0.  0.  0.]]


In [7]:
bcancer_data = datasets.load_breast_cancer()
print(bcancer_data.data.shape)
print(bcancer_data.target.shape)
print(bcancer_data.target_names)
print(bcancer_data.data[:1])
print(bcancer_data.feature_names)

(569, 30)
(569,)
['malignant' 'benign']
[[1.799e+01 1.038e+01 1.228e+02 1.001e+03 1.184e-01 2.776e-01 3.001e-01
  1.471e-01 2.419e-01 7.871e-02 1.095e+00 9.053e-01 8.589e+00 1.534e+02
  6.399e-03 4.904e-02 5.373e-02 1.587e-02 3.003e-02 6.193e-03 2.538e+01
  1.733e+01 1.846e+02 2.019e+03 1.622e-01 6.656e-01 7.119e-01 2.654e-01
  4.601e-01 1.189e-01]]
['mean radius' 'mean texture' 'mean perimeter' 'mean area'
 'mean smoothness' 'mean compactness' 'mean concavity'
 'mean concave points' 'mean symmetry' 'mean fractal dimension'
 'radius error' 'texture error' 'perimeter error' 'area error'
 'smoothness error' 'compactness error' 'concavity error'
 'concave points error' 'symmetry error' 'fractal dimension error'
 'worst radius' 'worst texture' 'worst perimeter' 'worst area'
 'worst smoothness' 'worst compactness' 'worst concavity'
 'worst concave points' 'worst symmetry' 'worst fractal dimension']


In [10]:
digits_X = digits_data.data
digits_y = digits_data.target
bcancer_X = bcancer_data.data
bcancer_y = bcancer_data.target

In [17]:
print(cross_val_score(BernoulliNB(), bcancer_X, bcancer_y, cv=3).mean())
print(cross_val_score(MultinomialNB(), bcancer_X, bcancer_y, cv=3).mean())
print(cross_val_score(GaussianNB(), bcancer_X, bcancer_y, cv=3).mean())

0.6274204028589994
0.8945790401930752
0.9367492806089297


In [16]:
print(cross_val_score(BernoulliNB(), digits_X, digits_y, cv=3).mean())
print(cross_val_score(MultinomialNB(), digits_X, digits_y, cv=3).mean())
print(cross_val_score(GaussianNB(), digits_X, digits_y, cv=3).mean())

0.8258236507780582
0.8708771489735053
0.8186003803550138


#### Вопрос 1

Каким получилось максимальное качество классификации на датасете breast_cancer?

GaussianNB: 0.9367492806089297

#### Вопрос 2

Каким получилось максимальное качество классификации на датасете digits?

MultinomialNB: 0.8708771489735053

#### Вопрос 3

Выберите верные утверждения и запишите их номера через пробел (в порядке возрастания номера):

1) На вещественных признаках лучше всего сработал наивный байесовский классификатор с распределением Бернулли

2) На вещественных признаках лучше всего сработал наивный байесовский классификатор с мультиномиальным распределением

3) Мультиномиальное распределение лучше показало себя на выборке с целыми неотрицательными значениями признаков

4) На вещественных признаках лучше всего сработало нормальное распределение

3, 4